cadence/transactions/fundFUSD.cdc

```diff
@@ -0,0 +1,52 @@
+import FUSD from 0xFUSDADDRESS
+import FungibleToken from 0xFUNGIBLETOKENADDRESS
+
+pub fun hasFUSD(_ address: Address): Bool {
+  let receiver = getAccount(address)
+    .getCapability<&FUSD.Vault{FungibleToken.Receiver}>(/public/fusdReceiver)
+    .check()
+
+  let balance = getAccount(address)
+    .getCapability<&FUSD.Vault{FungibleToken.Balance}>(/public/fusdBalance)
+    .check()
+
+  return receiver && balance
+}
+
+pub fun initFUSD(_ acct: AuthAccount) {
+  if acct.borrow<&FUSD.Vault>(from: /storage/fusdVault) == nil {
+    acct.save(<-FUSD.createEmptyVault(), to: /storage/fusdVault)
+  }
+  acct.unlink(/public/fusdReceiver)
+  acct.unlink(/public/fusdBalance)
+  acct.link<&FUSD.Vault{FungibleToken.Receiver}>(/public/fusdReceiver, target: /storage/fusdVault)
+  acct.link<&FUSD.Vault{FungibleToken.Balance}>(/public/fusdBalance, target: /storage/fusdVault)
+}
+
+transaction(address: Address, amount: UFix64) {
+  let tokenAdmin: &FUSD.Administrator
+  let tokenReceiver: &{FungibleToken.Receiver}
+
+  prepare(minterAccount: AuthAccount, receiverAccount: AuthAccount) {
```

Comment:
Also, the service account should already be initialized with an FUSD received because one is created when the FUSD contract is deployed. So I think you should be able to skip the init step for the service account (same as for FLOW).

Reply:
Good point about removing the service account initialization, I pushed an update.

By the separate step for initialization, do you mean I should add an initialize button instead of doing it automatically when funding? If we want to avoid the initialize step, another option is creating a separate transaction for the service account that doesn't initialize the receiver.